### PR TITLE
added stat 3.0 beta and added pygtk and dependent packages

### PR DESCRIPTION
**Pull request by lee218llnl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/dyninst/package.py [hunk 1]
```diff
@@ -55,17 +59,19 @@ def install(self, spec, prefix):
...
+                  '-DLIBDWARF_LIBRARIES=%s'
+                  % join_path(libdwarf.lib, 'libdwarf.so'),
                   *std_cmake_args)
 
             make()
             make("install")
 
-
-    @when('@:8.1')
+    @when('@:8.1')  # NOQA: ignore=F811
```
Comment (lee218llnl) [inline var/spack/repos/builtin/packages/dyninst/package.py]: This isn't passing through the checks:

var/spack/repos/builtin/packages/dyninst/package.py:74: [F811] redefinition of unused 'install' from line 53

It works when I run flake8 on the file on my local system. @tgamblin or others, and ideas on how to fix this?

**Comment by adamjstewart** [comment]
I started some of these packages in #1041, but these look way better. I never actually got mine to successfully install, so I'll remove my versions from that PR if I ever get back to it.

**Comment by adamjstewart** [comment]
Can you rebase this off of the latest develop? Some of the changes you've made to pcre and glib are already in develop.

**Comment by lee218llnl** [comment]
@adamjstewart I noticed those modifications and was planning a rebase. I think I successfully did a clean rebase (I've messed these up before), but the unit tests are failing here:

==> Error: the constraint 'None' matches multiple packages, and this is not allowed in this context
mpileaks@2.3%clang@3.3~debug~opt+shared+static arch=linux-debianwheezy-x86_64
mpileaks@2.3%clang@3.3~debug~opt+shared+static arch=linux-debianwheezy-x86_64
mpileaks@2.3%clang@3.3~debug~opt+shared+static arch=linux-debianwheezy-x86_64
==> Error: the constraint 'None' match no package, and this is not allowed in this context

I will see if anyone else has reported similar issues.

File: var/spack/repos/builtin/packages/glib/package.py [hunk 2]
```diff
@@ -30,11 +30,14 @@ class Glib(Package):
...
        and interfaces for such runtime functionality as an event loop,
        threads, dynamic loading and an object system."""
+
     homepage = "https://developer.gnome.org/glib/"
     url      = "http://ftp.gnome.org/pub/gnome/sources/glib/2.42/glib-2.42.1.tar.xz"
 
     version('2.49.4', 'e2c87c03017b0cd02c4c73274b92b148')
     version('2.42.1', '89c4119e50e767d3532158605ee9121a')
+    version('2.48.1', '67bd3b75c9f6d5587b457dc01cdcd5bb',
+            url='http://ftp.gnome.org/pub/GNOME/sources/glib/2.48/glib-2.48.1.tar.xz')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/glib/package.py]: You shouldn't need to specify the url for this one. Also, can you make sure the versions are in order from newest to oldest?

**Comment by lee218llnl** [comment]
@tgamblin @mplegendre @becker33 Can someone review and potentially merge this? I am hoping to have a non-beta STAT 3.0 soon, but in the meantime, I would like this merged before it gets stale.

**Pull request merged**